scripts/hand_test.py

```diff
@@ -1,81 +1,99 @@
 import rospy
-from sensor_msgs.msg import JointState
 from roboy_middleware_msgs.msg import MotorCommand
 import numpy as np
-from std_srvs.srv import Trigger
-from std_msgs.msg import Empty
 
-rospy.init_node("handshake")
-topic_root = '/roboy/brain/control/'
-publisher = rospy.Publisher(topic_root + "joint_targets", JointState, queue_size=1)
-msg = JointState()
+rospy.init_node("hand_finger_test")
+topic_root = '/roboy/pinky/'
 hand_msg = MotorCommand()
-hand_msg.global_id = [42,42,44,45]
-motorcmd_pub = rospy.Publisher(topic_root + "middleware/MotorCommand", MotorCommand, queue_size=1)
+motorcmd_pub = rospy.Publisher(topic_root + "middleware/MotorCommand", MotorCommand,
+                               queue_size=1)
 rate = rospy.Rate(100)
 
-def close_hand():
-    hand_msg.setpoint = [900]*len(hand_msg.global_id)
-    motorcmd_pub.publish(hand_msg)
-
-def open_hand():
-    hand_msg.setpoint = [0]*len(hand_msg.global_id)
-    motorcmd_pub.publish(hand_msg)
-
-def move_fingers(right_hand = True, targets = [0,0,0,0]):
+def move_fingers(right_hand=True, targets=None):
     """
-    :param target: array of finger target values between 0 (open hand) and 800 (closed hand)
-    lr: "left" or "right"
+    :param target: array of finger target values between 0 (open hand) and 1 (closed hand)
+    right_hand: True (right hand); False (left hand)
     left_ids = [38, 39, 40, 41]
     right_ids = [42, 43, 44, 45]
     """
+    if targets is None:
+        targets = [0, 0, 0, 0]
 
+    # just take the ids with actual values (>0)
+    right_ids = [42, 42, 44, 45]
+    left_ids = [38, 39, 40, 41]
     if right_hand:
-        hand_msg.global_id = [42, 42, 44, 45]
-
+        ids_rl = right_ids
     else:
-        hand_msg.global_id = [38, 39, 40, 41]
+        ids_rl = left_ids
+    targets_relevant = []
+    ids_picked = []
 
-    hand_msg.setpoint = [i*800 for i in targets]
+    for j in range(len(targets)):  # check if there are 0- values -> motor is not commanded
+        if targets[j] > 0.0:
+            ids_picked.append(ids_rl[j])
+            targets_relevant.append(targets[j])
+    hand_msg.global_id = ids_picked
+
+    # make sure values are within the limits
+    targets = clamp_values(targets_relevant, min_allowed_value=0.0, max_allowed_value=1.0)
+    hand_msg.setpoint = [i * 800 for i in targets]
     motorcmd_pub.publish(hand_msg)
-    rospy.sleep(0.07)
 
-def move_fingers_simultaneous(right_hand = True):
-    # hand finger moving test all fingers move the same
+def clamp_values(targets, min_allowed_value, max_allowed_value):
+    for t in range(len(targets)):
+        if targets[t] < min_allowed_value:
+            targets[t] = min_allowed_value
+            print("Warning: target value too low -> clamped it")
```

Comment:
```suggestion
            rospy.logwarn("Warning: target value too low -> clamped it")
```
use rospy logging instead of print in other parts as well